docs/plugin/errors.md

```diff
@@ -0,0 +1,506 @@
+
+```c++
+uint32_t error = Core::ERROR_TIMEDOUT;
+printf("Got error code %d (%s)\n", error, Core::ErrorToString(error));
+
+/* Output:
+Got error code 11 (ERROR_TIMEDOUT)
+*/
+```
+
+### COM-RPC Errors
+
+When designing an interface that will be exposed over COM-RPC, all functions should return a `Core::hresult` to indicate if the function executed successfully. On success, the function should return `Core::ERROR_NONE`. 
+
+Any data returned by the function should be stored in an output parameter instead of a return value. This ensures consistency across interfaces. If an error occurs over the COM-RPC transport or during marshalling/umarshalling the data, the most-significant bit will be used to indicate the error code is a COM error. 
+
+```cpp
+Core::hresult success = _remoteInterface->MyFunction();
+
+if (success != Core::ERROR_NONE) {
+	// An error occured, was this a result of the COM link or did the plugin return an error?
+	if (success & COM_ERROR == 0) {
+		printf("Plugin returned error %d (%s)\n", success, Core::ErrorToString(success));
+	} else {
+		printf("COM-RPC error %d (%s)\n", success, Core::ErrorToString(success));
+	}
+}
+```
+
+### JSON-RPC
+
+As with COM-RPC, JSON-RPC methods should return a `Core::hresult` value to indicate success or failure.  If the JSON-RPC method returns an error code other than `Core::ERROR_NONE`, it is treated as a failure.
+
+!!! note
+	Some older RDK plugins return a `success` boolean in their response to indicate errors. This is not recommended or necessary - simply return the appropriate error code from the method and a valid JSON-RPC error response will be generated.
+
+The returned JSON conforms to the [JSON-RPC 2.0](https://www.jsonrpc.org/) standard. In addition to the Thunder core error code, the response body may contain a JSON-RPC error as defined in the JSON-RPC specification
+
+| code             | message          | meaning                                                      |
+| ---------------- | ---------------- | ------------------------------------------------------------ |
+| -32700           | Parse error      | Invalid JSON was received by the server. An error occurred on the server while parsing the JSON text. |
+| -32600           | Invalid Request  | The JSON sent is not a valid Request object.                 |
+| -32601           | Method not found | The method does not exist / is not available.                |
+| -32602           | Invalid params   | Invalid method parameter(s).                                 |
+| -32603           | Internal error   | Internal JSON-RPC error.                                     |
+| -32000 to -32099 | Server error     | Reserved for implementation-defined server-errors.           |
+
+In the below example, an attempt is made in activate a non-existent plugin. The Controller plugin returns `ERROR_UNKNOWN_KEY`
```

Comment:
Your the native Englishman here but shouldn't it be "an attempt is made to activate" or "in activating" (feel free to laugh if I'm incorrect :) )

Reply:
Good spot, that's a typo :) Will correct!